docs/dwo-configuration.md

```diff
@@ -125,3 +125,83 @@ config:
 ```
 
 The config above will have newly created PVCs to have its access mode set to `ReadWriteMany`.
+
+## Configuring Custom Init Containers
+
+The DevWorkspace Operator allows cluster administrators to inject custom init containers into all workspace pods via the `config.workspace.initContainers` field in the global DWOC. This feature enables use cases such as:
+
+- Injecting organization-specific tools or configurations
+- Customizing the persistent home directory initialization logic
+- Extracting cluster utilities (e.g., `oc` CLI) to ensure version compatibility
+
+**Security Note:** Only trusted administrators should have RBAC permissions to edit the `DevWorkspaceOperatorConfig`, as custom init containers run in every workspace and can execute arbitrary code.
+
+### Basic Example: Injecting Custom Tools
+
+```yaml
+apiVersion: controller.devfile.io/v1alpha1
+kind: DevWorkspaceOperatorConfig
+metadata:
+  name: devworkspace-operator-config
+  namespace: $OPERATOR_INSTALL_NAMESPACE
+config:
+  workspace:
+    initContainers:
+      - name: inject-oc-cli
+        image: registry.redhat.io/openshift4/ose-cli:latest
+        command: ["/bin/sh", "-c"]
+        args:
+          - |
+            cp /usr/bin/oc /home/user/bin/oc
+            cp /usr/bin/kubectl /home/user/bin/kubectl
+        volumeMounts:
+          - name: persistent-home
+            mountPath: /home/user/
+```
+
+### Special Container: `init-persistent-home`
+
+A specially-named init container `init-persistent-home` can be used to override the built-in persistent home directory initialization logic when `config.workspace.persistUserHome.enabled: true`. This is useful for enterprises using customized UDI images that require different home directory setup logic.
+
+**Contract for `init-persistent-home`:**
+
+- **Name:** Must be exactly `init-persistent-home`
+- **Image:** Optional. If omitted, defaults to the first non-imported workspace container's image. If no suitable image can be inferred, the workspace will fail to start with an error.
+- **Command:** Optional. If omitted, defaults to `["/bin/sh", "-c"]`. If provided, must exactly match this value.
+- **Args:** Required. Must contain exactly one string with the initialization script.
```

Comment:
Is there a reason why we limit command and args for init persistent home container?

Reply:
IMHO for this feature I would prefer if there were no specific restrictions just for the persistent home init container to allow more freedom for configuration. If something is misconfigured the user would likely know immediately (maybe by an error when starting the workspace pod) when starting a workspace which I think is fine in this case

cc @tolusha @rohanKanojia

Reply:
I don't mind